File: fhc_2014/QR_A.py
# Facebook Hacker Cup 2014 Qualification Round
# A. Square Detector

import sys

def stripped(a):
  while a[0].strip('.') == '':
    del(a[0])
  while a[-1].strip('.') == '':
    del(a[-1])
  return a

def rotated(a):
  return ["".join(l) for l in zip(*[list(s) for s in a])]

def solve():
  N = int(sys.stdin.readline())
  a = []
  for i in range(N):
    a.append(sys.stdin.readline().rstrip())
  a = stripped(a)
  a = stripped(rotated(a))
  if len(a) != len(a[0]):
    return False
  if "".join(a).find('.') >= 0:
    return False
  return True

cases = int(sys.stdin.readline())
for case in range(cases):
  r = solve()
  s = "Case #" + str(case + 1) + ": "
  s += ["NO", "YES"][int(r)]
  print(s)

File: fhc_2020/QR_A.cpp
//
// Facebook Hacker Cup 2020 Qualification Round
// A. Travel Restrictions
//

#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

vector<string> solve(int n, string in, string out) {
	vector<string> ans(n, string(n, 'Y'));
	for (int i = 0; i < n; ++i) {
		for (int j = i - 1; j >= 0; --j) {
			if (out[j + 1] != 'Y' || in[j] != 'Y' || ans[i][j + 1] != 'Y') {
				ans[i][j] = 'N';
			}
		}
		for (int j = i + 1; j < n; ++j) {
			if (out[j - 1] != 'Y' || in[j] != 'Y' || ans[i][j - 1] != 'Y') {
				ans[i][j] = 'N';
			}
		}
	}
	return ans;
}

int main(int argc, char *argv[]) {
	int T, n;
	cin >> T;
	for (int t = 1; t <= T; ++t) {
		cin >> n;
		string in, out;
		cin >> in >> out;
		vector<string> ans = solve(n, in, out);
		cout << "Case #" << t << ":" << endl;
		for (string s : ans) {
			cout << s << endl;
		}
	}
	return 0;
}
